src/arrayMethodJoin.js

```diff
@@ -4,9 +4,29 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    if (!this.length) {
+      return '';
+    }
+
+    let result = String(this[0]);
+
+    if (this[0] === null || this[0] === undefined) {
+      result = '';
+    }
+
+    for (let i = 1; i < this.length; i++) {
+      if (separator === undefined) {
+        result += ',' + String(this[i]);
+      } else if (this[i] === null || this[i] === undefined) {
+        result += (String(separator) + '');
```

Comment:
```suggestion
        result += separator + '';
```
Не потрібно переводити сепаратор в рядок. Він і так є рядком